### DISCUSSION TITLE
Adding fixed width (fa-fw) class to button icon

### DISCUSSION THREAD

**Discussion by phill-85** [patternfly-yew/patternfly-yew#18] [General]
There a way of adding the font-awesome fixed width calss "fa-fw" to an icon on a button?

I've had a look at the icon code, but the macro code was a bit above my level of understanding.

**Comment by ctron** [comment]
Not sure how this works. If that would be an alternative to the `fas` class, then that would mean adding a new function (similar to `fas` and `pf`).

If that's a different concept, I am not sure.

**Comment by phill-85** [comment]
No, the fas class is still used, for example see:

`<button class="pf-c-button pf-m-plain" type="button" aria-label="Clear">
    <i class="fas fa-times fa-fw" aria-hidden="true"></i>
</button>`
 - https://www.patternfly.org/v4/components/search-input/html#no-match

**Comment by ctron** [comment]
I see, thx!

I am not sure the current `Icon` enum can properly represent this. I also think that there might be a better way than using an `enum`.

Ideally, one would write something like this:

```rust
html!(
  <>
    <Component icon={patternfly::Icon::SomeIcon}>
    <Component icon={fontawesome::Icon::Arrow}>
    <Component icon={fontawesome::Icon::Arrow.wrap(fontawesome::Variant::FixedWith)}>
    <Component icon={fontawesome::Icon::Other("foo-bar"))}>
  </>
)
```

So the `icon` attribute would merely be an `Html` type.

**Reply by phill-85** [reply]
How about creating an Icon component?
Something like?
    <Icon icon="Icon::fa-angle-right" fixed-width />
